### DISCUSSION TITLE
prevent the laptop from sleeping when closing the lid

### DISCUSSION THREAD

**Discussion by ithakaa** [newmarcel/KeepingYouAwake#200] [Ideas]
MacOS will not sleep if connected to power and an external monitor when the lid is closed

Once the power is removed, MacOS will automatically sleep

It would be a good idea to have an option to "not sleep" when "on battery" and connected to an external monitor when closing the lid.

**Comment by newmarcel** [comment]
Hi, this is a wish that comes up from time to time. By now I've added it to the [README](https://github.com/newmarcel/KeepingYouAwake?tab=readme-ov-file#does-it-work-on-a-macbook-with-a-closed-lid) because KeepingYouAwake does not and will not prevent system sleep when lid is closed.

To quote the README for future reference:
> **Does it work on a MacBook with a closed lid?**
> No. Because of _(sensible)_ macOS limitations and out of thermal considerations, KeepingYouAwake will only prevent sleep on desktop Macs and portable Macs with an open lid.

There might be different tools available that could satisfy this use case.
